### DISCUSSION TITLE
developing custom firmware, can I use x-min, and y-min without modification?

### DISCUSSION THREAD

**Discussion by drinnird** [FYSETC/FYSETC-E4#54] [Q&A]
Can I use x-min and y-min to attach sensors to fysetc e4 for use with a custom developed firmware?

![image](https://github.com/FYSETC/FYSETC-E4/assets/15637636/61c9a679-e2f5-4635-bb0c-a9f3ed6ba41f)

In this image you can see there is a 100r, 10k, and 100nf cap attached to both X min and Y min data lines to esp32. What do these do? Will they interfere with sensors connected to the ports?

If I want to remove the components to have direct access to the Pins 6 and 7 (gpio 34 & 35), which components are they on this schematic?

![image](https://github.com/FYSETC/FYSETC-E4/assets/15637636/4428aa59-768a-493b-9ffb-1966b2da9149)

**Comment by marcoa131** [comment]
The capacitors serve the purpose of 'debouncing' the pin, which entails regulating the rate at which voltage changes occur on that pin. Concurrently, the resistors facilitate controlled discharge of the capacitor when the pin is pulled low. These elements within the circuit are instrumental in attenuating potential noise originating from sources such as electromagnetic interference (EMI), floating switches, and power surges. To address your query, it is indeed possible to omit these components and directly engage the ESP pin's function. However, the necessity of doing so hinges on the specific objectives you aim to achieve.
